### PR TITLE
add otel with uptrace client

### PR DESCRIPTION
**Pull request by matthyx**
## Checklist before requesting a review
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] New and existing unit tests pass locally with my changes

**Please open the PR against the `dev` branch (Unless the PR contains only documentation changes)**

### CODE REVIEW

**Comment by dwertent** [comment]
@matthyx Is this PR ready for review?

**Comment by matthyx** [comment]
No I want to get the full spans first

File: sensor.go [hunk 1]
```diff
@@ -121,12 +131,16 @@ func filterNLogHTTPErrors(rw http.ResponseWriter, r *http.Request, next http.Han
...
 	}
 	if r.Body != nil {
-		io.Copy(ioutil.Discard, r.Body)
+		io.Copy(io.Discard, r.Body)
 	}
 }
 
 // main
 func main() {
+	// to enable otel, set OTEL_COLLECTOR_SVC=otel-collector:4317
```
Comment (dwertent) [inline sensor.go:140]: [TODO]: Get accountID and OTEL svc from env

**Review by dwertent — CHANGES_REQUESTED** [review]

File: sensor/verboseutils_test.go [hunk 2]
```diff
@@ -20,7 +21,7 @@ func Test_makeHostDirFilesInfo(t *testing.T) {
 	observedLogger := zap.New(observedZapCore)
```
Comment (matthyx) [inline sensor/verboseutils_test.go:21]: @dwertent the test failure comes from not being able to intercept the logger output... do you have any idea to fix that?

**Comment by matthyx** [comment]
@dwertent ready for review!

File: sensor.go [hunk 3]
```diff
@@ -32,7 +36,7 @@ func (blw bodyLogWriter) Write(b []byte) (int, error) {
 }
 
 var (
-	zapLogger *zap.Logger
+	zapLogger *otelzap.Logger
```
Comment (dwertent) [inline sensor.go:39]: [Question]: Why are we using here the `zapLogger`?

File: sensor.go [hunk 4]
```diff
@@ -56,13 +60,14 @@ func initLogger() *log.Logger {
 	// 	zapConf.ErrorOutputPaths = append(zapConf.ErrorOutputPaths, config.LogFileName)
 	// }
 
-	zapLogger, err = zapConf.Build()
+	l, err := zapConf.Build()
+	zapLogger = otelzap.New(l)
```
Comment (dwertent) [inline sensor.go:64]: [TODO]: The error check should happen before the initialization

File: sensor.go [hunk 5]
```diff
@@ -121,12 +132,22 @@ func filterNLogHTTPErrors(rw http.ResponseWriter, r *http.Request, next http.Han
...
+	if otelHost, present := os.LookupEnv("OTEL_COLLECTOR_SVC"); present {
+		ctx = logger.InitOtel("host-scanner",
+			os.Getenv(BuildVersion),
+			os.Getenv("ACCOUNT_ID"),
+			url.URL{Host: otelHost})
+		defer logger.ShutdownOtel(ctx)
+	}
+
 	fmt.Println("Starting Kubescape cluster node host scanner service")
 	fmt.Println("Host scanner service build version: " + BuildVersion)
```
Comment (dwertent) [inline sensor.go:152]:
```suggestion
	logger.L().Info("BuildVersion", helpers.String(BuildVersion))
```

**Review by dwertent — COMMENTED** [review]
Please review my changes. Nothing is blocking.

**Review by dwertent — APPROVED** [review]
Please see my comment from the previous review, they are not blocking.

**Pull request merged**